src/util/theme.py

```
@@ -11,7 +13,32 @@
+) -> Callable[Concatenate[Theme, P], R | None]:
+    def _fun(self: Theme, *args: P.args, **kwargs: P.kwargs) -> R | None:
+        filename, = args
+        assert isinstance(filename, str)
+        if not os.path.isfile(self._themepath(filename)):
+            return None
+        return fun(self, *args, **kwargs)
+    return _fun
+
+
+def _warn_resource_null[**P, R](
+    fn: Callable[Concatenate[ThemeSet, P], R],
```

Ruff (0.14.10)

Undefined name `ThemeSet`

(F821)